### PR TITLE
Retry

### PR DESCRIPTION
**Pull request by roykho**
* Add - Idempotency failure retry with easing to prevent multirequest issue.

### CODE REVIEW

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 1]
```diff
@@ -14,6 +14,17 @@
...
 
+	/**
+	 * Checks to see if request is invalid due to
+	 * multiple requests causing error.
+	 *
+	 * @since 4.0.5
+	 * @param array $error
+	 */
+	public function is_idempotency_error( $error ) {
+		return ( 'invalid_request_error' === $error->type && preg_match( '/There is currently another in-progress request using this Idempotent Key/', $error->message ) );
```
Comment (thorsten-stripe) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php]: Might be more robust to match on statusCode 409 instead, wdyt?

**Comment by roykho** [comment]
@thorsten-stripe ok so I actually changed the logic a bit. I am now not specifically looking for idempotency but any of the `invalid_request_error`, `idempotency_error` and `rate_limit_error` because those are all worth retrying with backoff. This may be better than status code.

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 2]
```diff
@@ -14,6 +14,21 @@
...
+	/**
+	 * Checks to see if request is invalid and that
+	 * they are worth retrying.
+	 *
+	 * @since 4.0.5
+	 * @param array $error
+	 */
+	public function is_retryable_error( $error ) {
+		return (
+			'invalid_request_error' === $error->type ||
```
Comment (thorsten-stripe) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php]: @roykho just a note that in most cases `invalid_request_error` is used for incorrect parameters in which case retrying with the same parameters will return the same error. However, we also use this for trying to charge sources that currently aren't in a chargeable state in which case retrying would make sense.

**Comment by thorsten-stripe** [comment]
LGTM
thanks for this!

**Pull request merged**